### ISSUE TITLE
Regarding generate method used in BART

### ISSUE THREAD

**Issue by kunalpagarey** [huggingface/transformers#4781]
# ❓ Questions & Help

## Details

I am new to NLP and text generation so please help me understand some basic things.
I fine-tuned BART on CNN/DM dataset using provide scripts in the examples section and it works fine.
I have some understanding of how model.generate() method works. But need to clarify some basic questions.
Is the generate() method used at the time of training/validation?
If yes, then how loss is computed if the beam size is greater than 1?
Or do we even use the parameters like in generate() method during training?

Please help. Thank you.

**Comment by patrickvonplaten** [comment]
Hey @kunalpagarey, 

The `.generate()` method cannot be used for training. It can be used for validation and testing. 
To create a good model for summarization the following components are important:

**Train**:
1. What model do you want to use? => Bart for example
2. What loss function do you want to use? => Usually people do pretraining and then finetuning on summarization using standard maximum likelihood, for finetuning see here: https://github.com/ohmeow/ohmeow_website/blob/master/_notebooks/2020-05-23-text-generation-with-blurr.ipynb

** Decoding method ** 
After having fine-tuned a model on summarization, the decoding method to apply is another whole question and is often done independently of training. Now, you can decide how to use the `generate()` method. You could for example try out a bunch of different hyperparameters for `.generate()` on your validation set and then decide on one setting for want to use for your test set. 
For more details on how to choose hyperparameters for `.generate()` check out: 
https://huggingface.co/blog/how-to-generate
